In [1]:
from astropy import units as u
from astropy import constants as const

### Maria's Homework

1. $\textbf{The Pulsar Population}$ The figure below shows the known pulsar population in P-$\dot{P}$ space.

In [4]:
P_NS = 10**0 * u.second
delP_NS = 10**(-15)

P_widow = 10**(-2.5) * u.second
delP_widow = 10**(-20)

(a) Calculate the minimum magnetic field strength of an average ordinary pulsar vs an average black
widow pulsar using:

$B(gauss) > 3.2 \times 10^{19}(\frac{P \dot{P}}{s})^{1/2}$

In [6]:
def mag_field_pulsar(P, delP):
    return 3.2 * 10**(19) * ((P * delP) / u.second)**(1/2) * u.Gauss

In [7]:
# Average Normal Pulsar
B_NS = mag_field_pulsar(P_NS, delP_NS)
B_NS

<Quantity 1.01192885e+12 G>

In [8]:
# Black Widow Pulsar
B_widow = mag_field_pulsar(P_widow, delP_widow)
B_widow

<Quantity 1.79949224e+08 G>

In [16]:
# Crab
mag_field_pulsar(0.0331*u.second, 4.22*10**(-13))

<Quantity 3.78198847e+12 G>

(b) Explain the differences in the magnetic field strengths of an ordinary pulsar vs a black widow pulsar,
if you find any variation at at all.


The minimum strength of an ordinary pulsar's magnetic field is $10^4$ times higher than that of a black widow pulsar. Both the period and the change of the spin rate are smaller for black widow pulsars, making their minimum magnetic field strength lower.

(c) Calculate the characteristic age of an average ordinary pulsar vs an average black widow pulsar.
Does this help explain what you find in part (a)? Use the following:

$\tau = \frac{P}{2 \dot{P}}$


In [9]:
def char_age_pulsar(P, delP):
    return P / (2 * delP)

In [14]:
# Average Normal Pulsar
t_NS = char_age_pulsar(P_NS, delP_NS)
t_NS.to(u.Myr)

<Quantity 15.84404391 Myr>

In [20]:
# Black Widow Pulsar
t_widow = char_age_pulsar(P_widow, delP_widow)
t_widow.to(u.Gyr)

<Quantity 5.01032661 Gyr>

In [19]:
# Crab
char_age_pulsar(0.0331*u.second, 4.22*10**(-13)).to(u.yr)

<Quantity 1242.74372825 yr>

The characteristic age of these pulsars assumes that there is no magnetic field decay, the initial spin period was much smaller, and assumes that the magnetic braking is like a magnet spinning in a vacuum. The characteristic age for a normal pulsar is much less than that of a black widow. Since there is magnetic field decay for black widow pulsars, the approximation is bad.

2. $\textbf{Pulsar Observations}$ The canonical equation used to determine the sensitivity achieved when detecting a pulsar is:

$S_{min} = \frac{\alpha \beta T_{sys}}{G (N \Delta v t_{int})^{1/2}} (\frac{W}{P − W})^{1/2} $

Here, α is the S/N, β accounts for processing losses, Tsys is the system temperature, G is antenna gain,
N is the number of polarizations, △v is the bandwidth, tint is the integration time, W is the pulse width,
and P is the pulse period. A smaller sensitivity is better. Let’s just call the first portion of this equation
C for now (C for we don’t Care) and focus on W and P.


(a) Calculate the minimum sensitivity achieved when detecting B0021-72C, a millisecond pulsar with P
= 0.005756 s and W = 0.664 ms vs the minimum sensitivity achieved when detecting J0021-0909
with P = 2.314 s and W = 23.8 ms (just using P and W, disregarding C). Which one would be easier
to detect?

In [27]:
def sensitivity(W, P):
    W = W.to(u.second)
    P = P.to(u.second)
    C = 1 # C for we don't Care
    return (W / (P - W))**(1./2.)

In [28]:
# J0021-0909 
P_J = 2.314 * u.second
W_J = 23.8 * u.millisecond
S_J = sensitivity(W_J, P_J)
S_J

<Quantity 0.10194168>

In [29]:
# B0021-72C
P_B = 0.005756 * u.second
W_B = 0.664 * u.millisecond
S_B = sensitivity(W_B, P_B)
S_B

<Quantity 0.36111027>

J0021-0909 would be easier to detect, as the sensitivity is smaller.

(b) Now pulling the integration time out of our C term, calculate the relative integration times between
B0021-72C vs J0021-0909.

In [31]:
# S = C * tint * #
# S_J / S_B = tint_J / tint_B
S_J / S_B

<Quantity 0.28230068>

In [32]:
S_B / S_J

<Quantity 3.54232226>

J0021-0909 would take 0.282 times less integration time than B0021-72C. Or, B0021-72C would take 3.5 times more time to detect than J0021-0909.

(c) In a few short sentences, describe the telescope you would design to detect millisecond pulsars if you
had unlimited resources.

The perfect telescope to detect millisecond pulsars would have the lowest system temperature possible, a high antenna gain, and a very large bandwidth.